kuksa_databroker/databroker/src/glob.rs

```diff
@@ -51,3 +52,113 @@ pub fn to_regex(glob: &str) -> Result<Regex, Error> {
     let re = to_regex_string(glob);
     Regex::new(&re).map_err(|_err| Error::RegexError)
 }
+
+fn has_asterisk_in_middle(input: &str) -> bool {
+    let input = input.trim(); // Remove leading and trailing whitespace
+    if input.starts_with('*') || input.ends_with('*') {
+        return false;
+    }
+
+    let asterisk_position = input.find('*');
+    if let Some(pos) = asterisk_position {
+        if input[pos - 1..].contains(".*.") {
+            return true;
+        }
+    }
+
+    false
+}
+
+pub fn to_regex_string_new(glob: &str) -> String {
+    let mut input = String::from(glob);
+
+    //check if asterisk in middle
+    if has_asterisk_in_middle(glob) {
+        input = glob.replace(".*.", ".");
+    }
+    // Construct regular expression
+
+    // Make sure we match the whole thing
+    // Start from the beginning
+    let mut re = String::from("^");
+
+    if input.ends_with(".*") {
```

Comment:
If it ends with `.*` shouldn't other replacements be done anyway?

Reply:
Following the use case "Vehicle.Cabin.Sunroof.*", if it is not replaced with ".[^.]+", it will return not only one-level child but any possible child, whose functionality belongs to the option "**".

Reply:
Ok.
But, the solution for that use case should not break other use cases.

My point is that the code that is meant to replace wildcards in the middle of the pattern is never run if `input.ends_with(".*")`.